src/types/dict.rs

```diff
@@ -430,7 +438,7 @@ enum DictIterImpl {
 
 impl DictIterImpl {
     #[inline]
-    fn next<'py>(
+    unsafe fn next_unchecked<'py>(
```

Comment:
We should document the safety invariant here.

(Also I assume you wanted to remove the critical section from this call?)

Reply:
Thanks for helping me spot the forest for the trees...